src/EnergyPlus/FluidProperties.cc

```diff
@@ -4487,6 +4487,137 @@ namespace FluidProperties {
 	//*****************************************************************************
 
 	Real64
+	GetSupHeatTempRefrig(
+		std::string const & Refrigerant, // carries in substance name
+		Real64 const Pressure, // actual pressure given as input
+		Real64 const Enthalpy, // actual enthalpy given as input
+		Real64 TempLow, // lower bound of temperature in the iteration
+		Real64 TempUp, // upper bound of temperature in the iteration
+		int & RefrigIndex, // Index to Refrigerant Properties
+		std::string const & CalledFrom // routine this function was called from (error messages)
+	)
+	{
+		// SUBROUTINE INFORMATION:
+		//       AUTHOR         Rongpeng Zhang
+		//       DATE WRITTEN   Jan 2016
+		//       RE-ENGINEERED  na
+
+		// PURPOSE OF THIS SUBROUTINE:
+		// Performs iterations to calculate the refrigerant temperature corresponding to the given 
+		// enthalpy and pressure.  Works only in superheated region.
+
+		// METHODOLOGY EMPLOYED:
+		// Perform iterations to identify the temperature by calling GetSupHeatEnthalpyRefrig.
+
+		// REFERENCES:
+		// na
+
+		// USE STATEMENTS:
+		// na
+
+		// Return value
+		Real64 ReturnValue;
+
+		// Locals
+		// FUNCTION ARGUMENT DEFINITIONS:
+
+		// FUNCTION PARAMETERS:
+		// the enthalpy calculated from the pressure found
+		static std::string const RoutineName( "GetSupHeatTempRefrig: " );
+		static std::string const RoutineNameNoSpace( "GetSupHeatTempRefrig:" );
+
+		// INTERFACE BLOCK SPECIFICATIONS:
+		// na
+
+		// DERIVED TYPE DEFINITIONS:
+		// na
```

Comment:
As an FYI, several of these comment sections aren't relevant and not needed.  I think they remain in the old Fortran template, but in general, especially for new functions, don't add them.  As another point of guidance, moving forward we'll want local variables defined as close as possible to where they are used/needed.  No need to define the locals at the top of the function like we used to do in Fortran.

This function doesn't appear to be a great candidate to become a member function of a class, but I'll be looking for that as I comment along here.